httpClients/boot-restclient/src/main/java/com/example/restclient/bootrestclient/services/PostService.java

```diff
@@ -1,81 +1,63 @@
 package com.example.restclient.bootrestclient.services;
 
-import com.example.restclient.bootrestclient.exception.MyCustomRuntimeException;
 import com.example.restclient.bootrestclient.model.response.PostDto;
 import java.util.List;
+import java.util.Map;
 import java.util.Optional;
 import org.springframework.core.ParameterizedTypeReference;
-import org.springframework.http.HttpStatusCode;
-import org.springframework.http.MediaType;
+import org.springframework.http.HttpMethod;
 import org.springframework.stereotype.Service;
-import org.springframework.web.client.RestClient;
 
 @Service
 public class PostService {
```

Comment:
@coderabbitai, can you please generate junits for this class using @AutoConfigureMockRestServiceServer and @SpringBootTest